lib/luacrypto_rng.c

```diff
@@ -0,0 +1,199 @@
+/*
+* SPDX-FileCopyrightText: (c) 2025 jperon <cataclop@hotmail.com>
+* SPDX-License-Identifier: MIT OR GPL-2.0-only
+*/
+
+/***
+* Low-level Lua interface to the Linux Kernel Crypto API for synchronous
+* Random Number Generators (RNG).
+*
+* This module provides a `new` function to create RNG objects,
+* which can then be used for generating random bytes.
+*
+* @module crypto_rng
+*/
+
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <crypto/rng.h>
+#include <linux/err.h>
+#include <linux/limits.h>
+#include <linux/slab.h>
+
+#include <lua.h>
+#include <lualib.h>
+#include <lauxlib.h>
+#include <lunatik.h>
+
+#include "luacrypto.h"
+
+LUNATIK_PRIVATECHECKER(luacrypto_rng_check, struct crypto_rng *);
+
+LUACRYPTO_RELEASER(rng, struct crypto_rng, crypto_free_rng, NULL);
+
+/***
+* RNG object methods.
+* These methods are available on RNG objects created by `crypto_rng.new()`.
+* @see crypto_rng.new
+* @type crypto_rng
+*/
+
+/***
+* Generates a specified number of random bytes, with optional seed.
+* @function crypto_rng:generate
+* @tparam integer num_bytes The number of random bytes to generate.
+* @tparam[opt] string seed Optional seed material to mix into the RNG. If nil or omitted, no explicit seed is used.
+* @treturn string A binary string containing the generated random bytes.
+* @raise Error on failure (e.g., allocation error, crypto API error).
+*/
+static int luacrypto_rng_generate(lua_State *L)
+{
+	struct crypto_rng *tfm = luacrypto_rng_check(L, 1);
+	size_t num_bytes = (size_t)luaL_checkinteger(L, 2);
+	luaL_argcheck(L, num_bytes >= 0, 2, "number of bytes must be non-negative");
+
+	size_t seed_len = 0;
+	const char *seed_data = lua_tolstring(L, 3, &seed_len);
+
+	luaL_Buffer b;
+	char *buffer = luaL_buffinitsize(L, &b, num_bytes);
+
+	lunatik_try(L, crypto_rng_generate, tfm, seed_data, (unsigned int)seed_len, (u8 *)buffer, (unsigned int)num_bytes);
+
```

Comment:
```suggestion
```

Reply:
it's not resolved, right? I think it's weird to have this \n here but don't in here https://github.com/luainkernel/lunatik/pull/293/files#diff-3b16446e204a14703776af98f9dad9896259ee837085ccea480f47403f508d58R87